src/library/scala/Enumeration.scala

```diff
@@ -188,7 +188,13 @@ abstract class Enumeration (initial: Int) extends Serializable {
   protected final def Value(i: Int, name: String): Value = new Val(i, name)
 
   private def populateNameMap(): Unit = {
-    val fields: Array[JField] = getClass.getDeclaredFields
+    @tailrec def getFields(clazz: Class[_], acc: Array[JField]): Array[JField] = {
+      if (clazz.getSuperclass == null)
```

Comment:
Maybe just a style thing, but I'd suggest `if clazz == null` for termination.
```
getFs(clazz, acc) = if (clazz == null) acc else getFs(clazz.getSuperclass, if (clazz.getDFs.isEmpty) acc else acc ++ clazz.getDFs)
getFs(getClass.getsuperclass, getClass.getDeclaredFields)
```
which will normally allocate one array? Actually not sure yet if `++` is optimal for empty thing to append.

Reply:
I guess `appendedAll` means always make a copy and not return this if appending an empty thing.